src/Algolia.Search/Models/Search/Query.cs

```diff
@@ -61,27 +68,27 @@ public Query(string searchQuery = null)
         /// Filter hits by facet value.
         /// </summary>
         [JsonConverter(typeof(FiltersConverter))]
-        public IEnumerable<IEnumerable<string>> FacetFilters { get; set; }
+        public List<List<string>> FacetFilters { get; set; }
 
         /// <summary>
         /// Create filters for ranking purposes, where records that match the filter are ranked highest.
         /// </summary>
         [JsonConverter(typeof(FiltersConverter))]
-        public IEnumerable<IEnumerable<string>> OptionalFilters { get; set; }
+        public List<List<string>> OptionalFilters { get; set; }
 
         /// <summary>
         /// Filter on numeric attributes.
         /// </summary>
         [JsonConverter(typeof(FiltersConverter))]
-        public IEnumerable<IEnumerable<string>> NumericFilters { get; set; }
+        public List<List<string>> NumericFilters { get; set; }
 
         /// <summary>
         /// Filter hits by tags.
         /// tagFilters is a different way of filtering, which relies on the _tags attribute. It uses a simpler syntax than filters.
         /// You can use it when you want to do simple filtering based on tags.
         /// </summary>
         [JsonConverter(typeof(FiltersConverter))]
-        public IEnumerable<IEnumerable<string>> TagFilters { get; set; }
+        public List<List<string>> TagFilters { get; set; }
```

Comment:
Why did we change all `*Filters` types to List instead of `IEnumerable`?

Reply:
At time I made these changes `System.Text.Json` was not supporting `IEnumerable` yet and more generally Interfaces IIRC, that's why I had to specify an implem.

Maybe it has been fixed since then.